tests/test_languages.py

```diff
@@ -2168,3 +2171,30 @@ def test_validate_extra_keys_when_invalid(self, lang_id, lang_info, log_msg):
         result = self.validator._validate_extra_keys(lang_id, lang_info)
         self.assertEqual(log_msg, self.get_log_str())
         self.assertFalse(result)
+
+    @parameterized.expand([
+        param(date_string='3 de marzo 2019', languages=["en"], settings={
+            "DEFAULT_LANGUAGES": ["es"]
+        }, expected=datetime(2019, 3, 3, 0, 0)),
+    ])
+    def test_parse_settings_default_languages(self, date_string, languages, settings, expected):
+        result = parse(date_string, languages=languages, settings=settings)
+        assert result == expected
+
+    @parameterized.expand([
+        param(date_string='RANDOM_WORD ', settings={
+            "DEFAULT_LANGUAGES": ["en"]
+        })
+    ])
+    def test_parse_settings_default_languages_no_language_detect(self, date_string, settings):
+        result = parse(date_string, settings=settings)
+        assert result is None
+
+    @parameterized.expand([
+        param(date_string='12/12/2000', languages=["es"], settings={
+            "DEFAULT_LANGUAGES": ["en"]
+        }),
+    ])
+    def test_parse_settings_default_languages_with_detected_language(self, date_string, languages, settings):
+        result = parse(date_string, languages=languages, settings=settings)
+        assert type(result) == datetime
```

Comment:
For this test I would use a string recognized by the language in `languages` and not a generic date. For example:

```suggestion
    @parameterized.expand([
        param(date_string='29 mai 2021', languages=["fr"], settings={
            "DEFAULT_LANGUAGES": ["en", "es"]
        }),
    ])
    def test_parse_settings_default_languages_with_detected_language(self, date_string, languages, settings):
        result = parse(date_string, languages=languages, settings=settings)
        assert type(result) == datetime
```

And I would add the expected date instead of just checking the type. :)

Reply:
Could you add a test for `DEFAULT_LANGUAGES` with:
* `DateDataParser` (and `get_date_data`)
* `search_dates`

Reply:
Hi @noviluni, I have added tests for both the functions but `search_dates` can you please recheck as search dates only select only one language

Reply:
Hey removing this file `dateparser_data/language_detection_models/__init__.py` is leading to errors and other folders like `dateparser_data`

Pl suggest

Reply:
Hi @gavishpoddar, good job with the tests!

> Hey removing this file dateparser_data/language_detection_models/__init__.py is leading to errors and other folders like dateparser_data contains empty init.py

The empty __init__.py it's not an issue, but that folder (`language_detection_models`) is not used. Could you explain me which errors are you having and/or how to reproduce them? I was able to remove it and run the tests and all is working.  :thinking: 

I was able to directly remove it, but if you have som git related errors, you can try with `git rm`. Give me more details, please :)

Reply:
I was having  issue 

`ModuleNotFoundError: No module named 'langdetect.detector_factory'; 'langdetect' is not a package
`

I have removed that and instead imported used `langdetect.detector_factory.DetectorFactory`

Reply:
Now its solved

Reply:
Hey @gavishpoddar, I think there was a confusion. I asked to remove the 

![image](https://user-images.githubusercontent.com/22377678/131301766-7d83ba2e-f0ab-4247-ac64-b8ca230880b1.png)

Not the:

![image](https://user-images.githubusercontent.com/22377678/131301821-25192060-bac7-4f01-8c6b-843dc3e28d82.png)

You can restore the second one and remove the first one.

Reply:
Done 👍🏻, thanks